### PR TITLE
[ReputationOracle] Refactor Auth 

### LINKED ISSUE #1685: Reputation Oracle - Authentication refactor
**Description**
Refactor Reputation Oracle authentication to work as authentication in Job Launcher

**Motivation**
Improve the code quality and security

### PR DESCRIPTION
**Pull request by mrhouzlane**
## Description

Refactor Reputation Oracle authentication to work as authentication in Job Launcher

## Summary of changes

Follow Job Launcher authentication. 

Closes #1685

### CODE REVIEW

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -83,13 +111,18 @@ export class AuthService {
     return this.auth(userEntity);
   }
 
-  public async signup(data: UserCreateDto): Promise<UserEntity> {
+  public async signup(data: UserCreateDto, ip?: string): Promise<UserEntity> {
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:114]: add hcaptcha token verification even though you leave it commented

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -111,90 +144,117 @@ export class AuthService {
...
 
-  public async restorePassword(data: RestorePasswordDto): Promise<boolean> {
-    const tokenEntity = await this.tokenRepository.findOne({
-      uuid: data.token,
-      tokenType: TokenType.PASSWORD,
-    });
+  public async restorePassword(
+    data: RestorePasswordDto,
+    ip?: string,
+  ): Promise<void> {
```
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:246]: add hcaptcha token verification even though you leave it commented

**Review by portuu3 — CHANGES_REQUESTED** [review]
you need to add the migrations to apply the changes in the database

File: packages/apps/reputation-oracle/server/src/modules/auth/strategy/jwt.http.ts [hunk 3]
```diff
@@ -4,66 +4,43 @@ import { Injectable, Req, UnauthorizedException } from '@nestjs/common';
...
   ) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
       ignoreExpiration: false,
-      secretOrKey: configService.get<string>(
-        ConfigNames.JWT_SECRET,
-        'secretkey',
-      ),
+      secretOrKey: configService.get<string>(ConfigNames.JWT_PUBLIC_KEY, ''),
+      algorithm: ['RS256'],
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/strategy/jwt.http.ts:22]: removed

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.controller.ts [hunk 4]
```diff
@@ -80,8 +115,8 @@ export class AuthJwtController {
     status: 404,
     description: 'Not Found. Could not find the requested content.',
   })
-  public signin(@Body() data: SignInDto): Promise<AuthDto> {
-    return this.authService.signin(data);
+  public signin(@Body() data: SignInDto, @Ip() ip: string): Promise<AuthDto> {
+    return this.authService.signin(data, ip);
   }
 
   @Public()
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.controller.ts:122]: changed

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.module.ts [hunk 5]
```diff
@@ -23,20 +24,27 @@ import { Web3Module } from '../web3/web3.module';
       inject: [ConfigService],
       imports: [ConfigModule],
       useFactory: (configService: ConfigService) => ({
-        secret: configService.get<string>(ConfigNames.JWT_SECRET, 'secretkey'),
+        privateKey: configService.get<string>(ConfigNames.JWT_PRIVATE_KEY),
+        publicKey: configService.get<string>(ConfigNames.JWT_PUBLIC_KEY),
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.module.ts:28]: removed

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts [hunk 6]
```diff
@@ -309,14 +376,18 @@ export class AuthService {
...
+      nonce: nonce,
+    };
 
     const userEntity = await this.userService.createWeb3User(
+      web3UserCreateDto,
       data.address,
-      data.type,
     );
 
-    await kvstore.set(data.address, 'ACTIVE');
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:318]: restored.

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts [hunk 7]
```diff
@@ -50,23 +58,43 @@ export class AuthService {
...
+      1000,
+    );
+    this.keyLength = this.configService.get<number>(
+      ConfigNames.APIKEY_KEY_LENGTH,
+      64,
+    );
   }
 
-  public async signin(data: SignInDto): Promise<AuthDto> {
+  public async signin(data: SignInDto, ip?: string): Promise<AuthDto> {
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:97]: added and commented

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts (hunk 1 above)
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:114]: added and commented

File: packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts (hunk 2 above)
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/modules/auth/auth.service.ts:246]: added and commented

File: packages/apps/reputation-oracle/server/.dockerignore
Comment (portuu3) [inline packages/apps/reputation-oracle/server/.dockerignore:1]: undo this

File: packages/apps/reputation-oracle/server/src/database/migrations/1710942293785-RefactorAuth.ts
Comment (portuu3) [inline packages/apps/reputation-oracle/server/src/database/migrations/1710942293785-RefactorAuth.ts:1]: This is not right, this is creating the entire database. A migrations should only add or modify what is new from the previous migration

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/reputation-oracle/server/src/database/migrations/1710942293785-RefactorAuth.ts
Comment (mrhouzlane) [reply packages/apps/reputation-oracle/server/src/database/migrations/1710942293785-RefactorAuth.ts:1]: ok, updated

**Review by portuu3 — APPROVED** [review]

**Pull request merged**